### PR TITLE
nix(arm64): make nix setup ARM64 compatible

### PR DESCRIPTION
**Pull request by gsebil08**
## Problem

Current nix setup don't build on ARM64 Mac 🍎 

## Solution

~~Add `system.aarch64-darwin` and make `dream` "buildable"~~
Add `dream` within the overlay.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -18,29 +16,30 @@
...
+    { self, nixpkgs, flake-utils, nix-npm-buildpackage, ocaml-overlays }:
     with flake-utils.lib;
-    eachSystem [ system.x86_64-linux ]
-      (system:
-        let
-          pkgs = import nixpkgs {
-            inherit system;
-            overlays = [ ocaml-overlays.overlay (import ./nix/overlay.nix) ];
-          };
+    eachSystem [ system.x86_64-linux system.aarch64-darwin ] (system:
```
Comment (gsebil08) [inline flake.nix:22]: This is the only change in this file, the rest is formatting with `nixfmt`

File: nix/deku.nix [hunk 2]
```diff
@@ -44,9 +42,8 @@ pkgs.ocamlPackages.buildDunePackage {
     reason
   ]);
 
-  propagatedBuildInputs = [
-    npmPackages
-  ];
+  propagatedBuildInputs = [ npmPackages ]
+    ++ (with pkgs.ocamlPackages; [ ppx_inline_test ]);
```
Comment (gsebil08) [inline nix/deku.nix:46]: Only change here.

File: nix/overlay.nix [hunk 3]
```diff
@@ -1,42 +1,45 @@
 final: prev:
```
Comment (gsebil08) [inline nix/overlay.nix:1]: No change here, only formatting.

File: nix/shell.nix [hunk 4]
```diff
@@ -10,18 +10,18 @@ pkgs.mkShell {
     export USE_NIX=y
```
Comment (gsebil08) [inline nix/shell.nix:10]: No change neither.

File: nix/overlay.nix [hunk 5]
```diff
@@ -1,42 +1,49 @@
...
+let disableCheck = package: package.overrideAttrs (o: { doCheck = false; });
+in {
+  ocamlPackages = prev.ocaml-ng.ocamlPackages_5_00.overrideScope'
+    (oself: osuper: rec {
+      rely = osuper.reason-native.rely.overrideAttrs (_: {
+        postPatch = ''
+          substituteInPlace src/rely/TestSuiteRunner.re --replace "Pervasives" "Stdlib"
+        '';
+      });
+      dream = osuper.dream.overrideAttrs (o: {
```
Comment (gsebil08) [inline nix/overlay.nix:11]: Only real change

File: nix/deku.nix [hunk 6]
```diff
@@ -19,14 +20,11 @@ pkgs.ocamlPackages.buildDunePackage {
...
 
-  nativeBuildInputs = [
-    nodejs
-  ] ++ (with pkgs.ocamlPackages; [
+  nativeBuildInputs = [ nodejs ] ++ (with pkgs.ocamlPackages; [
     cmdliner
     ppx_deriving
     ppx_deriving_yojson
     lwt
-    dream
```
Comment (gsebil08) [inline nix/deku.nix:29]: This is an actual change

**Comment by gsebil08** [comment]
Working on debian
KO on macos with M1...

```shell script
$ nix build .#deku
warning: Using saved setting for 'trusted-substituters = https://cache.nixos.org/ https://deku.cachix.org' from ~/.local/share/nix/trusted-settings.json.
error: builder for '/nix/store/3vnjap7s0jid8yi3k812gapik83732np-ocaml5.00.0+trunk-dream-1.0.0-alpha2.drv' failed with exit code 1;
       last 10 log lines:
       >   [OK]          headers         20   with_header update.
       >
       > Full test results in `/private/tmp/nix-build-ocaml5.00.0+trunk-dream-1.0.0-alpha2.drv-0/source/_build/default/test/unit/_build/_tests/Dream'.
       > Test Successful in 0.004s. 34 tests run.
       > File "test/expect/eml/dune", line 10, characters 1-15:
       > 10 |  (inline_tests)
       >       ^^^^^^^^^^^^^^
       > (cd _build/default/test/expect/eml && .test_expect_eml.inline-tests/inline_test_runner_test_expect_eml.exe inline-test-runner test_expect_eml -source-tree-root ../../.. -diff-cmd -)
       > Command got signal BUS.
       >
       For full logs, run 'nix log /nix/store/3vnjap7s0jid8yi3k812gapik83732np-ocaml5.00.0+trunk-dream-1.0.0-alpha2.drv'.
error: 1 dependencies of derivation '/nix/store/r0viy5cv8gcrgz7c7s03gcqma6iq8666-ocaml5.00.0+trunk-sidechain-0.0.0-dev.drv' failed to build
```

**Comment by gsebil08** [comment]
I am feeling so dumb... Thanks @ulrikstrid!
I didn't even think about adding `doCheck = false` within the `dream` of `nix/overlay.nix`... Working perfectly now!

**Review by ulrikstrid — APPROVED** [review]
LGTM, @anchpop could you try this as well?

**Review by d4hines — APPROVED** [review]
LGTM

**Comment by anchpop** [comment]
@ulrikstrid for some reason I'm still not able to get nix to work properly on my mac. Please don't block this PR on my account :p

**Pull request merged**